test/remote/README.md

```diff
@@ -0,0 +1,240 @@
+```
+
+Set the admin password to use:
+
+```shell
+export ARGOCD_E2E_ADMIN_PASSWORD=$(kubectl get secrets argocd-initial-admin-secret -o jsonpath='{.data.password}'|base64 -d)
+```
+
+Run the tests
+
+```shell
+./test/remote/run-e2e-remote.sh make test-local
+```
+
+### On remote non-OpenShift cluster
+
+In another shell, do a port-forward to the API server's service:
+
+```shell
+kubectl -n argocd-e2e port-forward svc/argocd-server 443:4443
+```
+
+Set Argo CD Server endport:
+
+```shell
+export ARGOCD_SERVER=127.0.0.1:4443
+```
+
+Set the admin password to use:
+
+```shell
+export ARGOCD_E2E_ADMIN_PASSWORD=$(kubectl get secrets argocd-initial-admin-secret -o jsonpath='{.data.password}'|base64 -d)
+```
+
+Run the tests
+
+```shell
+./test/remote/run-e2e-remote.sh make test-local
+```
+
+### On remote OpenShift cluster
+
+You should first scale down GitOps Operator, since it will revert changes made
+during the tests instantly:
+
+```shell
+oc -n openshift-operators scale deployment --replicas=0 gitops-operator
+```
+
+Set the endpoint by using the route created by the operator:
+
+```shell
+export ARGOCD_SERVER=$(oc -n argocd-e2e get routes argocd-test-server -o jsonpath='{.spec.host}')
+```
+
+Set the admin password, created by the operator:
+
+```shell
+export ARGOCD_E2E_ADMIN_PASSWORD=$(oc -n argocd-e2e get secrets argocd-test-cluster -o jsonpath='{.data.admin\.password}' | base64 -d)
+```
+
+Set the name of the Argo CD instance as installed by the operator (below example assumes operand name of `argocd-test`):
+
+```shell
+export ARGOCD_E2E_NAME_PREFIX=argocd-test
+```
+
+Run the tests with currently known failing tests disabled:
+
+```shell
+./test/remote/run-e2e-remote.sh make test-local ARGOCD_E2E_SKIP_OPENSHIFT=true
+```
+
+## Running single or multiple tests isolated
+
+This should be run in the same shell where you set `ARGOCD_SERVER` and `ARGOCD_E2E_ADMIN_PASSWORD` variables
+
+1. Run the `go test` command through the wrapper, e.g. to run a test named `MyTestName`:
+
+   ```
+   $ ./test/remote/run-e2e-remote.sh go test -v github.com/argoproj/argo-cd/test/e2e -run ^MyTestName$^
+   ```
+
+
+## Further configuration
+
+Some environment variables can control the behavior of the tests:
+
+* `ARGOCD_SERVER` - the remote endpoint of the Argo CD API server to use for tests
+* `ARGOCD_E2E_ADMIN_PASSWORD` - the admin user's password to use
+* `ARGOCD_E2E_TEST_TIMEOUT` - timeout for the complete test suite, specified as duration (e.g. `2h` or `1h30m`)
+* `ARGOCD_E2E_DEFAULT_TIMEOUT` - timeout in seconds for each context operation (e.g. sync). Default `30`.
+* `ARGOCD_E2E_NAMESPACE` - the namespace where Argo CD is running in for the tests
+* `ARGOCD_E2E_NAME_PREFIX` - if your Argo CD installation has a name prefix (e.g. installed by the Operator), specify it here
+
+Furthermore, you can skip various classes of tests by setting the following to true:
+
+```
+# GnuPG features not yet available with GitOps Operator
```

Comment:
Probably good enough to mention that the following may be disabled optionally if support for the same is not available in your installation of ArgoCD.

Reply:
Agreed. Rephrased sentence.